src/providers/hana/qgshanaconnectionpool.h

```diff
@@ -23,44 +23,48 @@
 #include <memory>
 #include <QMutex>
 
-inline QString qgsConnectionPool_ConnectionToName( QgsHanaConnection *c )
-{
-  return c->connInfo();
-}
-
-inline void qgsConnectionPool_ConnectionCreate( const QgsDataSourceUri &uri, QgsHanaConnection *&c )
-{
-  c = QgsHanaConnection::createConnection( uri );
-}
-
-inline void qgsConnectionPool_ConnectionDestroy( QgsHanaConnection *c )
-{
-  delete c;
-}
-
-inline void qgsConnectionPool_InvalidateConnection( QgsHanaConnection *c )
-{
-  Q_UNUSED( c );
-}
-
-inline bool qgsConnectionPool_ConnectionIsValid( QgsHanaConnection *c )
-{
-  Q_UNUSED( c );
-  return true;
-}
-
 class QgsHanaConnectionPoolGroup
   : public QObject, public QgsConnectionPoolGroup<QgsHanaConnection *>
 {
     Q_OBJECT
 
   public:
-    explicit QgsHanaConnectionPoolGroup( const QString &name );
+    explicit QgsHanaConnectionPoolGroup( const QString &name )
+      : QgsConnectionPoolGroup<QgsHanaConnection*>( name )
+    {
+      initTimer( this );
+    }
+
+    ~QgsHanaConnectionPoolGroup() override
+    {
+      for ( const Item &item : std::as_const( mConnections ) )
+      {
+        delete item.connection;
```

Comment:
Just wondering if for this kind of loops a simple call to qDeleteAll would be a better choice https://doc.qt.io/qt-6/qtalgorithms.html#qDeleteAll-1

Reply:
Notice we delete the element connection inside Item, running qDeleteAll on the whole struct Item will not compile.

Reply:
of course, sorry for the noise.